xbmc/platform/posix/filesystem/PosixDirectory.cpp

```diff
@@ -34,67 +36,67 @@ bool CPosixDirectory::GetDirectory(const CURL& url, CFileItemList &items)
 {
   std::string root = url.Get();
 
+  constexpr std::array<std::string_view, 3> exclusions = {".", "..", "lost+found"};
+  const auto notExcluded = exclusions.end();
+
   if (IsAliasShortcut(root, true))
     TranslateAliasShortcut(root);
 
+  if (!URIUtils::HasSlashAtEnd(root))
+    root += "/";
+
   DIR *dir = opendir(root.c_str());
   if (!dir)
     return false;
 
+  struct stat buffer;
+
+  std::vector<CFileItemPtr> fileItems;
   struct dirent* entry;
   while ((entry = readdir(dir)) != NULL)
   {
-    if (strcmp(entry->d_name, ".") == 0 || strcmp(entry->d_name, "..") == 0)
-      continue;
-
-    std::string itemLabel(entry->d_name);
-    CCharsetConverter::unknownToUTF8(itemLabel);
-    CFileItemPtr pItem(new CFileItem(itemLabel));
-    std::string itemPath(URIUtils::AddFileToFolder(root, entry->d_name));
-
-    bool bStat = false;
-    struct stat buffer;
+    const std::string& strName = entry->d_name;
 
-    // Unix-based readdir implementations may return an incorrect dirent.d_ino value that
-    // is not equal to the (correct) stat() obtained one. In this case the file type
-    // could not be determined and the value of dirent.d_type is set to DT_UNKNOWN.
-    // In order to get a correct value we have to incur the cost of calling stat.
-    if (entry->d_type == DT_UNKNOWN || entry->d_type == DT_LNK)
+    if (std::ranges::find(exclusions, strName) == notExcluded)
     {
-      if (stat(itemPath.c_str(), &buffer) == 0)
-        bStat = true;
-    }
+      std::string itemPath(root + strName);
```

Comment:
Is this equivalent to `URIUtils::AddFileToFolder`? Can there be edge cases where a simple concatenation is wrong?